src/bermudan.cpp

```diff
@@ -162,8 +165,8 @@ Rcpp::List bermudanSwaptionEngine(Rcpp::List rparam,
     QuantLib::Rate dummyFixedRate = 0.03;
     boost::shared_ptr<QuantLib::IborIndex> indexSixMonths(new QuantLib::Euribor6M(rhTermStructure));
         
-    QuantLib::Date startDate = calendar.advance(settlementDate, 1, QuantLib::Years, floatingLegConvention);
-    QuantLib::Date maturity = calendar.advance(startDate, 5, QuantLib::Years, floatingLegConvention);
+    //QuantLib::Date startDate = calendar.advance(settlementDate, 1, QuantLib::Years, floatingLegConvention);  //took out hard coded 
+    //QuantLib::Date maturity = calendar.advance(startDate, 5, QuantLib::Years, floatingLegConvention);         //dates
```

Comment:
Yes, it is a tradeoff.   It wasn't immediately clear _where_ the backwards compatibility breaking change occurred.  Here we could argue it is a bug fix.

We can have it both ways.  As I understand it, `startDate` and `maturity` should be but were not parameters.  They were constants here.   So we could (should ?) make them parameters, and consider defaulting them to 1 and 5.   We can always do that with a warning a nice idiom is something like

```r
f <- function(par) {
    if (missing(par)) {     # now we know
         par <- 42
         warning("You did not set par when calling f.  Setting default of 42.")
   }
   # get on with function ...
}
```

Not saying we must, just saying we can and that it would make sense to me.  You are deeper in that code.

As for 'users must specify' versus 'sensible defaults' I generally like the latter....